src/components/date-time-picker/date-time-picker.component.ts

```diff
@@ -203,12 +201,38 @@ export class DateTimePickerComponent implements AfterViewInit, OnDestroy {
         public datepicker: DateTimePickerService,
         @Optional() private _rangeService: DateRangeService,
         @Optional() private _rangeOptions: DateRangeOptions) {
+    }
 
-        datepicker.selected$.pipe(distinctUntilChanged(dateComparator), takeUntil(this._onDestroy))
-            .subscribe(date => this.dateChange.emit(date));
+    ngOnInit(): void {
 
-        datepicker.timezone$.pipe(distinctUntilChanged(timezoneComparator), takeUntil(this._onDestroy))
-            .subscribe((timezone: DateTimePickerTimezone) => this.timezoneChange.emit(timezone));
+        // if there is no defined timezone then try and use the system default, otherwise fallback to
+        // offset of 0, e.g. GMT or user defined equivalent
+        if (!this.timezone) {
+            this.datepicker.timezone$.next(this.datepicker.getCurrentTimezone());
+        }
+
+        this.datepicker.selected$.pipe(distinctUntilChanged(dateComparator), takeUntil(this._onDestroy))
+            .subscribe(date => {
+                if (!this.date && !date || dateComparator(this.datepicker.date$.value, date)) {
+                    return;
+                }
+                this.dateChange.emit(date);
+            });
+
+        this.datepicker.timezone$.pipe(takeUntil(this._onDestroy))
+            .subscribe((timezone: DateTimePickerTimezone) => {
+                if (!this.datepicker.timezone$.value && timezone || !timezoneComparator(this.datepicker.timezone$.value, timezone)) {
+                    this.timezone = timezone;
+                }
+            });
+
+        this.datepicker.timezone$.pipe(distinctUntilChanged(timezoneComparator), takeUntil(this._onDestroy))
+            .subscribe((timezone: DateTimePickerTimezone) => {
+                if (!this.timezone && !timezone || timezoneComparator(this.datepicker.timezone$.value, timezone)) {
+                    return;
+                }
+                this.timezoneChange.emit(timezone);
+            });
```

Comment:
I don't think this is right, we need to emit on user interaction with the component, not programmatic input. Did you try binding to the spinner outputs like I suggested?

Reply:
The spinner outputs are already linked. When the spinners are increased/decreased the selected$ observable is passed the new value in the time-view component on valueChange().

Reply:
This will emit on programmatic change of `timezone`, look above at:
```ts
    @Input() set timezones(value: DateTimePickerTimezone[]) {
        if (value !== undefined) {
            this.datepicker.timezones$.next(value);
        }
    }
```

The spinners may be linked to `selected$`, but that doesn't help with emitting `timezoneChange`. Most of these observables are updated on both user and programmatic changes, so you can't really emit in subscriptions. I would say to hook those spinner change events up to a new `timezoneChange` output on `TimeViewComponent` (you would need to convert from string to `DateTimePickerTimezone`), then hook those up in `DateTimePickerComponent` to emit `timezoneChange` to the consuming application. There are other ways to do this, if this doesn't work, but I think it's at least the main part of the solution.

Reply:
Added this in